### DISCUSSION TITLE
Building a new cluster with talosctl gen config results in 127.0.0.1 defined as the endpoints in the talosconfig file

### DISCUSSION THREAD

**Discussion by digilink** [siderolabs/talos#8297] [Q&A] [answered]
I am just getting started with talos, so I don't know a ton about how things work or are expected to work. I was following the getting started guide here to install on a vm from an iso, I'm doing a single node cluster for a POC: 

https://www.talos.dev/v1.6/introduction/getting-started/

When I run talosctl gen config talos-test https://10.23.40.1:6443, I see this in the resulting talosconfig file: 

```
context: talos-test
contexts:
    talos-test:
        endpoints:
            - 127.0.0.1
```

I would have expected the endpoints key to be set to 10.23.40.1 vs. 127.0.0.1? Is this a bug or expected behavior? I was able to build the cluster, but to use the dashboard correctly as an example, I had to change this key to the correct IP address.

**Comment by steverfrancis** [comment, marked as answer]
That's as expected.
10.23.40.1 may be a Talos VIP, which should not be used as an endpoint (as it will be down when K8s or etcd is down, and then you couldn't issue talosctl commands to recover.)
More details: https://www.talos.dev/v1.6/learn-more/talosctl/#endpoints-and-nodes

In your case, just configure talosconfig with the correct endpoints: https://www.talos.dev/v1.6/introduction/prodnotes/#endpoints
or pass them in to the command.

**Reply by sanmai-NL** [reply]
I've read the docs, understand your answer to some extent, but this still doesn't make enough sense to me.

First of all, I can't find info about Talos Linux virtual IP functionality other than some platform-specific tutorials, not on how they would be tied to the state of the Kubernetes control plane. A VIP can be created on a lower level, no?

When one generates a config, no assumptions should be made about such a crucial config key like `talosctl` endpoints. Putting an IPv4 link-local address serves what use-case exactly? I can only imagine, when the admin has some sort of (SSH) tunnel to an endpoint on their system.

**Reply by steverfrancis** [reply]
Good point. 
from your example command, 10.23.40.1 should not be the default endpoint as it could (should, for production) be a VIP (talos or external to the cluster).
(and re Talos VIP being tied to K8s state: see https://www.talos.dev/v1.6/talos-guides/network/vip/#requirements
"Note that the virtual IP election depends on etcd being up, as Talos uses etcd for elections and leadership (control) of the IP address.")

But that doesn't explain why loopback address is in there. It may make more sense to have that not defined at all, and have talosctl explicitly prompt if it is undefined. I'll check with engineering..

**Reply by steverfrancis** [reply]
https://github.com/siderolabs/talos/issues/8500
